presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/StaticHiveCluster.java

```diff
@@ -67,17 +73,22 @@ public HiveMetastoreClient createMetastoreClient()
         List<HostAndPort> metastores = new ArrayList<>(addresses);
         Collections.shuffle(metastores.subList(1, metastores.size()));
 
-        TTransportException lastException = null;
+        TException lastException = null;
         for (HostAndPort metastore : metastores) {
             try {
-                return clientFactory.create(metastore);
+                HiveMetastoreClient hiveMetastoreClient = clientFactory.create(metastore);
+                if (!com.google.common.base.Strings.isNullOrEmpty(hiveUserName)) {
+                    hiveMetastoreClient.setUGI(hiveUserName, new ArrayList<String>());
```

Comment:
does this method needs to take a list and to return a list, it looks like you are ignoring these

Reply:
This method is to tell Metastore who I am and next time I access Metastore through thrift, Metastore will use UGI.doAs () ...